### ISSUE TITLE
Writing CHARMM RESIdues to stream file

### ISSUE THREAD

**Issue by jchodera** [ParmEd/ParmEd#1103]
ParmEd can [read `RESI`due templates from RTF files](https://github.com/ParmEd/ParmEd/blob/9b8711893cca81ed4a0a942053be8381664b3995/parmed/charmm/parameters.py#L822) but the `CharmmParameterSet.write()` method [claims to write RTF](https://github.com/ParmEd/ParmEd/blob/9b8711893cca81ed4a0a942053be8381664b3995/parmed/charmm/parameters.py#L1030-L1044) residues but appears not to actually have the capability of writing `RESI`due templates.

Am I missing something, or does this still need to be implemented?

**Comment by swails** [comment]
It looks like this needs to be implemented.  It's been awhile since I worked on this, but as I recall, the RTF files were only used by CHARMM to build PSF files -- that is, given a PSF, PRM, and *just the atom types* in the RTF CHARMM can run a simulation.  At this point, I believe CHARMM supports putting the atom information needed for simulation in the PRM file, so RTFs aren't needed for simulating at all.

*Writing* files needed only for system construction in both Amber and CHARMM formats never seemed that important to me.  Amber was written around, and for, the Amber force field and doesn't even support writing all of the functional forms that the simulation engines themselves support (e.g., the CHARMM force field).  So translating <anything> -> Amber didn't make sense.  CHARMM was similar.  The force fields you would usually use CHARMM to build the system "started out" as CHARMM files to begin with.  If you wanted to use CHARMM on a different force field, you'd build it in its "native" suite and simply use ParmEd to create a PSF + PRM file.

OpenMM is different -- it doesn't have its "own" force fields like CHARMM + AMBER do, so it made sense to write force field translators to OMM formats (that and your lab donated time, expertise, and code to make it happen).

What is your use-case here for writing RESI records to RTF files?

**Comment by jchodera** [comment]
We are trying to support Wonpil Im's CHARMM-GUI by enabling the openforcefield toolkit to parameterize small molecules in user systems and export templates with parameters via CHARMM stream (str) files. We can currently 
* Build an OpenMM System
* Load this as a Structure
* Convert to a CharmmParameterSet
* Append ResidueTemplates constructed from the Structure 
* Export to stream

but this last step is missing the RTF residue templates.

This would help us deploy to an extremely large audience, so it would be an incredibly useful feature. Wonpil says that he only needs ATOM, BOND, and IMPR cards---the IC cards are optional.

**Comment by swails** [comment]
Makes sense.  I'll see what I can do, but I probably won't have the time to get to it for a couple weeks.
